iOS Client/PodcastsController/PodcastsCoordinator.swift

```diff
@@ -38,15 +47,77 @@ class PodcastsCoordinator: Coordinator, StoreSubscriber {
                 }
             }
         }
-        
+    }
+    
+    @objc func updateDuration(_ sender: Timer) {
+        if let totalDuration = player.currentItem?.duration.seconds {
+            playerDelegate?.updateCurrentDuration(current: getCurrentDuration(), total: totalDuration)
+        }
     }
     
     // MARK: - StoreSubscriber
     func newState(state: MBAppState) {
+        // Handle changes in the player state
+        switch state.podcastsState.player {
+        case .initialized:
+            break
+        case .playing:
+            if let podcast = state.podcastsState.selectedPodcast, let guid = podcast.guid, let url = URL(string: guid) {
+                if currentPlayingPodcast?.guid != guid {
+                    let item = AVPlayerItem(url: url)
+                    player.replaceCurrentItem(with: item)
+                }
+                player.play()
+                currentPlayingPodcast = state.podcastsState.selectedPodcast
+                timer?.invalidate()
+                timer = Timer.scheduledTimer(timeInterval: 1.0, target: self, selector: #selector(updateDuration(_:)), userInfo: nil, repeats: true)
+                print("playing podcast")
+            }
+        case .paused:
+            player.pause()
+            timer?.invalidate()
+        case .error, .finished:
+            timer?.invalidate()
+            break
+        }
+        
+        // Handle changes in navigation state
         guard state.navigationState.selectedTab == .podcasts, let newRoute = state.navigationState.routes[.podcasts] else {
             return
         }
         build(newRoute: newRoute)
         route = newRoute
+        playerDelegate?.handler = self
```

Comment:
confusing. This should be something more like podcastDetailViewController.delegate = self
 

maybe if let vc = navigationController.viewControllers.last as? PodcastDetailViewController {
  vc.delegate = self
}